x/x.go

```diff
@@ -78,8 +78,8 @@ const (
 	// ErrorNoData is an error returned when the requested data cannot be returned.
 	ErrorNoData = "ErrorNoData"
 	// ValidHostnameRegex is a regex that accepts our expected hostname format.
-	ValidHostnameRegex = "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]" +
-		"|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])$"
+	ValidHostnameRegex = `^([a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62}){1}(\.[a-zA-Z0-9_]{1}` +
```

Comment:
Do we need this when we can just use `net.SplitHostPort` which returns an error?

Reply:
Ignore this